src/libutils/oglapphelpers/msl.mm

```diff
@@ -0,0 +1,504 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright Contributors to the OpenColorIO Project.
+
+#include <unordered_map>
+#include <string>
+#include <sstream>
+#include <iostream>
+
+#include <OpenColorIO/OpenColorIO.h>
+
+#include "msl.h"
+#include "mtltexture.h"
+
+#import <Metal/Metal.h>
+
+namespace OCIO_NAMESPACE
+{
+
+std::vector<float> RGB_to_RGBA(const float* lutValues, int valueCount)
+{
+    if(valueCount % 3 != 0)
+        throw Exception("Value count should be divisible by 3.");
+    
+    valueCount = valueCount * 4 / 3;
+    std::vector<float> float4AdaptedLutValues;
+    if(lutValues != nullptr)
+    {
+        float4AdaptedLutValues.resize(valueCount);
+        const float *rgbLutValuesIt = lutValues;
+        float *rgbaLutValuesIt = float4AdaptedLutValues.data();
+        const float *end = rgbaLutValuesIt + valueCount;
+            
+        while(rgbaLutValuesIt != end) {
+            *rgbaLutValuesIt++ = *rgbLutValuesIt++;
+            *rgbaLutValuesIt++ = *rgbLutValuesIt++;
+            *rgbaLutValuesIt++ = *rgbLutValuesIt++;
+            *rgbaLutValuesIt++ = 1.0f;
+        }
+    }
+    
+    return float4AdaptedLutValues;
+}
```

Comment:
[Comment] The memory management of the image data introduces at least one useless copy. As `float4AdaptedLutValues` is a local variable the return could make a copy. With C++11 one could use `std::move` -- or -- the new buffer could be an in/out parameter of the method!